clients/da-vinci-client/src/main/java/com/linkedin/davinci/kafka/consumer/StoreIngestionTask.java

```diff
@@ -3078,6 +3080,18 @@ private void processStartOfPush(
       ControlMessage controlMessage,
       PartitionConsumptionState partitionConsumptionState) {
     StartOfPush startOfPush = (StartOfPush) controlMessage.controlMessageUnion;
+    if (partitionConsumptionState.isEndOfPushReceived()) {
+      LOGGER.warn(
+          "Received START_OF_PUSH after END_OF_PUSH for replica: {}. This may indicate out-of-order or "
+              + "duplicate control messages. Details: Incoming SoP: {}, Previous SoP timestamp: {}, Previous "
+              + "EoP timestamp: {}. The message will be ignored.",
```

Comment:
>Previous SoP timestamp: {} 

Maybe I should call it last SoP timestamp for the storeVersion on this host?

Reply:
Either way is fine... as you wish!

Reply:
I missed your comment.  😄